ios/sdk/src/CallKit.m

```diff
@@ -0,0 +1,334 @@
+    CXSetMutedCallAction *action
+    = [[CXSetMutedCallAction alloc] initWithCallUUID:uuid muted:muted];
+    CXTransaction *transaction = [[CXTransaction alloc] initWithAction:action];
+    [self requestTransaction:transaction resolve:resolve reject:reject];
+}
+
+// Start outgoing call
+RCT_EXPORT_METHOD(startCall:(NSString *)uuidString
+                     handle:(NSString *)handle
+                      video:(BOOL)video
+                    resolve:(RCTPromiseResolveBlock)resolve
+                     reject:(RCTPromiseRejectBlock)reject)
+{
+#ifdef DEBUG
+    NSLog(@"[RNCallKit][startCall] uuidString = %@", uuidString);
+#endif
+    NSUUID *uuid = [[NSUUID alloc] initWithUUIDString:uuidString];
+    CXHandle *callHandle
+        = [[CXHandle alloc] initWithType:CXHandleTypeGeneric value:handle];
+    CXStartCallAction *action
+        = [[CXStartCallAction alloc] initWithCallUUID:uuid handle:callHandle];
+    action.video = video;
+    CXTransaction *transaction = [[CXTransaction alloc] initWithAction:action];
+    [self requestTransaction:transaction resolve:resolve reject:reject];
+}
+
+// Indicate call failed
+RCT_EXPORT_METHOD(reportCallFailed:(NSString *)uuidString
+                  resolve:(RCTPromiseResolveBlock)resolve
+                  reject:(RCTPromiseRejectBlock)reject)
+{
+    NSUUID *uuid = [[NSUUID alloc] initWithUUIDString:uuidString];
+    [callKitProvider reportCallWithUUID:uuid
+                            endedAtDate:[NSDate date]
+                                 reason:CXCallEndedReasonFailed];
+    resolve(nil);
+}
+
+// Indicate outgoing call connected
+RCT_EXPORT_METHOD(reportConnectedOutgoingCall:(NSString *)uuidString
+                  resolve:(RCTPromiseResolveBlock)resolve
+                  reject:(RCTPromiseRejectBlock)reject)
+{
+    NSUUID *uuid = [[NSUUID alloc] initWithUUIDString:uuidString];
+    [callKitProvider reportOutgoingCallWithUUID:uuid
+                                connectedAtDate:[NSDate date]];
+    resolve(nil);
+}
+
+// Update call in case we have a display name or video capability changes
+RCT_EXPORT_METHOD(updateCall:(NSString *)uuidString
+                     options:(NSDictionary *)options
+                     resolve:(RCTPromiseResolveBlock)resolve
+                      reject:(RCTPromiseRejectBlock)reject)
+{
+#ifdef DEBUG
+    NSLog(@"[RNCallKit][updateCall] uuidString = %@ options = %@", uuidString, options);
+#endif
+    NSUUID *uuid = [[NSUUID alloc] initWithUUIDString:uuidString];
+    CXCallUpdate *update = [[CXCallUpdate alloc] init];
+    if (options[@"displayName"]) {
+        update.localizedCallerName = options[@"displayName"];
+    }
+    if (options[@"hasVideo"]) {
+        update.hasVideo = [(NSNumber*)options[@"hasVideo"] boolValue];
+    }
+    [callKitProvider reportCallWithUUID:uuid updated:update];
+    resolve(nil);
+}
+
+#pragma mark - Helper methods
+
+- (CXProviderConfiguration *)getProviderConfiguration:(NSDictionary* )settings
```

Comment:
There're multiple names for one and the same thing - configuration, options, settings - which is not in accord with our coding style.